settings/Core.setting.php

```diff
@@ -1062,4 +1062,23 @@
     'description' => ts('Acceptable Mime Types that can be used as part of file urls'),
     'help_text' => NULL,
   ],
+  'http_timeout' => [
+    'group_name' => 'CiviCRM Preferences',
+    'group' => 'core',
+    'name' => 'http_timeout',
+    'type' => 'Integer',
+    'quick_form_type' => 'Element',
+    'html_type' => 'text',
+    'html_attributes' => [
+      'size' => 2,
+      'maxlength' => 3,
+    ],
+    'default' => 5,
+    'add' => '5.15',
+    'title' => ts('HTTP request timeout'),
+    'is_domain' => 1,
+    'is_contact' => 0,
+    'description' => ts('How long should HTTP requests through Guzzle application run for in seconds'),
```

Comment:
I'm a bit torn about mentioning guzzle here rather than being more generic - but it could iterate I guess